Source: Erik Li <li.erik@gmail.com>
Upstream: Patch submitted <http://sourceforge.net/tracker/?func=detail&aid=2892724&group_id=15273&atid=115273>
Reason: Fixes fuser kill functionality after introduction of special process types with pid 0 in fuser 22.8


--- src/fuser.c.orig	2009-11-02 20:20:06.775137993 -0500
+++ src/fuser.c	2009-11-02 23:50:01.629976909 -0500
@@ -1452,11 +1452,13 @@
 	struct procs *pptr;
 
 	for (pptr = proc_head; pptr != NULL; pptr = pptr->next) {
-		if ((opts & OPT_INTERACTIVE) && (ask(pptr->pid) == 0))
-			continue;
-		if (kill(pptr->pid, sig_number) < 0) {
-			fprintf(stderr, _("Could not kill process %d: %s\n"),
-				pptr->pid, strerror(errno));
+		if ( pptr->proc_type == PTYPE_NORMAL ){
+			if ((opts & OPT_INTERACTIVE) && (ask(pptr->pid) == 0))
+				continue;
+			if ( kill(pptr->pid, sig_number) < 0) {
+				fprintf(stderr, _("Could not kill process %d: %s\n"),
+						pptr->pid, strerror(errno));
+			}
 		}
 
 	}
